app/components/add_course_button.html.erb

```diff
@@ -0,0 +1,26 @@
+<% if required_organisation_details_present? %>
+  <%= govuk_button_link_to(
+    "Add course",
+    new_publish_provider_recruitment_cycle_course_path(
+      provider_code: provider.provider_code,
+      recruitment_cycle_year: provider.recruitment_cycle_year
+    ),
+    class: "govuk-!-margin-bottom-6"
+  ) %>
+
+<% else %>
+
+<div class="govuk-inset-text">
+    <p class="govuk-body">Before adding a course, you need to:</p>
+      <ul class="govuk-list govuk-list--bullet">
+
+        <% incomplete_sections_hash.each do |section, path| %>
+          <% if send(section) %>
```

Comment:
I'd change here to something like:

```
   incomplete_sections.each do |section|
      govuk_link_to("Add #{section.name}", section.path
   end
```
The method incomplete_sections would return a more rich object but just selecting the sections that are incomplete.

The advantage is that the code becomes cleaner and easy to understand (you won't have to call #send in the view for example) and also move some methods of the component to the rich object.

Reply:
Done it like this, WDYT? 

https://github.com/DFE-Digital/publish-teacher-training/pull/3697/commits/a11cc1c47c580db96d407ca5a84b667e7ef0003d

Reply:
Much better! I will have only a small tweak. Excellent work mate!

Reply:
🙇🏼‍♀️